src/cli/parse-flags.ts

```diff
@@ -287,13 +292,43 @@ interface CLIArgValue {
 }
 
 /**
- * When a parameter is set in the format `--foobar=12` at the CLI (as opposed to
- * `--foobar 12`) we want to get the value after the `=` sign
+ * Parse an 'equals' argument, which is a CLI argument-value pair in the
+ * format `--foobar=12` (as opposed to a space-separated format like
+ * `--foobar 12`).
+ *
+ * To parse this we split on the `=`, returning the first part as the argument
+ * name and the second part as the value. We join the value on `"="` in case
+ * there is another `"="` in the argument.
+ *
+ * This function is safe to call with any arg, and can therefore be used as
+ * an argument 'normalizer'. If CLI argument is not an 'equals' argument then
+ * the return value will be a tuple of the original argument and an empty
+ * string `""` for the value.
+ *
+ * In code terms, if you do:
  *
- * @param commandArgument the arg in question
- * @returns the value after the `=`
+ * ```ts
+ * const [arg, value] = parseEqualsArg("--myArgument")
+ * ```
+ *
+ * Then `arg` will be `"--myArgument"` and `value` will be `""`, whereas if
+ * you do:
+ *
+ *
+ * ```ts
+ * const [arg, value] = parseEqualsArg("--myArgument=myValue")
+ * ```
+ *
+ * Then `arg` will be `"--myArgument"` and `value` will be `"myValue"`.
+ *
+ * @param arg the arg in question
+ * @returns a tuple containing the arg name and the value (if present)
  */
-const getEqualsValue = (commandArgument: string) => commandArgument.split('=').slice(1).join('=');
+export const parseEqualsArg = (arg: string): [string, string] => {
```

Comment:
I refactored this function to return a tuple of the argument name (parsed from the CLI input) and the value it is set to if present. The value will be present if an argument like `--foobar=bop` is passed, but not if arg like `--foobar` is passed.

The first value in the tuple will be `--foobar` for both of those examples, so even for non-'equals sign' type CLI args this function be be safely used to normalize the argument names, so that we can count on things like

```ts
parseEqualsArg("--foobar")[0] === parseEqualsArgs("--foobar=baz")[0]
```
being true